### DISCUSSION TITLE
Taking Documentation Requests

### DISCUSSION THREAD

**Discussion by thabaum** [oqtane/oqtane.framework#3800] [General]
Hello Oqtane Community 🚀 

I am just putting out notice I am currently taking any documentation requests relating to my upcoming push to enhance them.

My focus next week is to spend at least 2 full -10 hour days worth of time on guides for getting started with using Linux and Windows operating systems, Apache, Nginx, and IIS web services along with the 4 supported SQL server types supported (MySQL, SQLite, MSSQL and LocalDb).  Anything the community would love for me to be sure to include I am all ears for the next week relating this area of the current Oqtane Framework documentation.  

The next direction I will moving towards is enhancing Module and Theme development.  However with all the latest changes in Oqtane Framework new render modes we may divert more time to going that direction until that future void gets filled, then work towards the goals I have outlined above.  A majority of latest documentation updates coming with the next version of Oqtane will be relating to administration and development.  These areas can be refined as we review the new documentation to include from the upcoming release notes and whatever Blog/Wiki articles will follow to help explain what's new.

If you have any specific things you feel can use some improvements in documentation and do not feel like opening an issue please feel free to let us know here in this thread.  I will be putting some serious time in next week to enhance documentation pushing out some [Oqtane.Docs](https://github.com/oqtane/oqtane.docs) repository pull requests to review prior the upcoming documentation meeting.  
___

FYI the documentation meeting is open to anyone who wants to help contribute and is currently being held at **8AM PST, 11AM EST, on the 4th Wednesday** of each month.  Our meeting discussion is located in this [link](https://github.com/oqtane/oqtane.docs/discussions/22).  Let @iJungleboy know if you would like to attend by sending him your contact info and he will forward you the meeting details.  It is not as technical of a meeting as the developer meetings.  Both administrators and developers are all encouraged to join up.

I ask anyone willing to help by posting some helpful must have steps or information to be sure to include in this thread or a [discussion](https://github.com/oqtane/oqtane.docs/discussions) or [issue](https://github.com/oqtane/oqtane.docs/issues) in [Oqtane.Docs](https://github.com/oqtane/oqtane.docs) (preferred).  

A link to a discussion on something that was helpful to you that you had to dig for since it is not included in our docs or any articles and you want to help the next person out, this would be a perfect example of a contribution to toss our way for review.  Another great example is mentioning a crucial step missing that may have been overlooked, is new, or needs better clarification by walking through these guides and providing us with feedback.  Even just mention the word documentation and one day one of us can review that thread and push out a docs pr if missing or found useful to provide to others.

I am confident all this effort will help the community in the long run.  I look forward to meeting any new contributors helping generate pull requests and/or providing valuable feedback.

Thank you.

Cheers!

**Comment by ASADeveloper** [comment]
It would indeed be nice if there was some documentation for a framework that is now at version 5!
I'm sure Shaun knows how important documentation is to adoption and building a community. That's why I find it strange that I can only find one dated Oqtane Module Development series on Youtube and a few (also dated) articles on BlazorHelpWebsite about Oqtane.
Although the mentioned resources are still partly applicable, it does not promote Oqtane enough to get more people excited about the project.
The problem with documentation is that if there is no documentation, others cannot learn and contribute to more documentation.

Thanks for your effort.

**Reply by sbwalker** [reply]
@ASADeveloper you are absolutely correct - documentation helps drive adoption. This is exactly why @thabaum and others are trying to accelerate efforts in this area. There has been a lot of progress on https://docs.oqtane.org in recent months and I am hopeful this trend continues.

**Reply by thabaum** [reply]
@ASADeveloper thanks for the feedback, it helps gauge where the efforts need to be for people like myself trying to contribute.  

@sbwalker is working on a community Wiki along with more Blog articles to come for sure as we have made a push for 5.1 recently that paid off huge. 

 @iJungleboy is ready to take on what we can throw his way for 5.1 updates along with his own efforts to get a handle this recent push on this rapidly evolving and maturing development framework.  
 
One of my contributions I am hoping to produce after the additional getting started guides is to focus on module and theme development using Oqtane Framework.  

There are still a few things to see play out along with community enhancements that are produced, but at this point I am very excited to have something like this to build upon today.  With the right feedback and support provided by our community this can all become very satisfying and a must for everyone to visit to get a hold of all the latest key concepts with each release.

Oqtane.Docs will be a greater resource, however the community Wiki and website Blog will allow the community and maintainer to provide a bit more color to each topic.  Really I would be happy to categorize these blogs in the Oqtane Docs Guides, ones that can be found to be very helpful relating to concepts.

Might I point out we will be starting the most active new year for Oqtane on record as far as the GitHub graph shows.  I believe this momentum is just the beginning as there are more exciting updates to come that should help in terms of performance.  This weekend I will make my first real attempts at providing some new documentation.   Oqtane.Docs has also seen a major boost since new year as well and I am going to see to it that some of this momentum continues with an if I can, as I can attitude.

Hopefully things go well as I learn from this documentation experience and allows me to more easily provide more like you are requesting.  This effort of mine will include ways to help enhance the docs relating to being able to contribute as well.

I see the effort into the docs so far we have been fortunate enough to have as it helps, and I want to give back some in this area is all.  It takes teamwork to make a dream work and Rome wasn't built overnight!

[Oqtane Documentation Meeting ](https://github.com/oqtane/oqtane.docs/discussions/22) next Wednesday so I will do my best to get as much prepared prior to then. This is what we have used to provide some additional guidance on contributing information and the discussion board is also for providing some additional ideas and context.  It would be exciting to see one or two more "go getters" get on board here for the fun of it.  I am trying to do my best here to fill that noticeable void.  It is not easy to move time this direction... i get it :)

My prediction is by 5.1.2 we will be up to speed, this includes more guides as we do up some drafts on more technically advanced use cases in an organized place in docs.  Especially for working with render modes.

Thanks again.

Cheers!
## :rocket:

**Reply by iJungleboy** [reply]
@thabaum totally agree with what you say.

What's important though is that we do things step-by-step, so better publish things we already documented (eg. the install docs you're working on) and not make the task so big that it never completes 😉

**Reply by GregStevenson** [reply]
I also concur.
